pyaedt/generic/farfield_visualization.py

```diff
@@ -0,0 +1,2187 @@
+        farfield_data["Gain"] = gain
+        farfield_data["Gain_Total"] = gain
+        farfield_data["Gain_dB"] = 10 * np.log10(gain)
+        gain = 2 * np.pi * np.abs(np.power(farfield_data["rETheta"], 2)) / accepted_power / 377
+        farfield_data["Gain_Theta"] = gain
+        gain = 2 * np.pi * np.abs(np.power(farfield_data["rEPhi"], 2)) / accepted_power / 377
+        farfield_data["Gain_Phi"] = gain
+
+        directivity = 2 * np.pi * np.abs(np.power(farfield_data["rETotal"], 2)) / radiated_power / 377
+        farfield_data["Directivity"] = directivity
+        farfield_data["Directivity_Total"] = directivity
+        farfield_data["Directivity_dB"] = 10 * np.log10(directivity)
+        directivity = 2 * np.pi * np.abs(np.power(farfield_data["rETheta"], 2)) / radiated_power / 377
+        farfield_data["Directivity_Theta"] = directivity
+        directivity = 2 * np.pi * np.abs(np.power(farfield_data["rEPhi"], 2)) / radiated_power / 377
+        farfield_data["Directivity_Phi"] = directivity
+
+        # farfield_data["Element_Location"] = port_positions
+        return farfield_data
+
+    @pyaedt_function_handler()
+    def get_accepted_power(self):
+        """Compute the accepted power from active s-parameters and incident power.
+
+        Returns
+        -------
+        float
+            Total accepted power.
+        """
+
+        if self.active_s_parameters is not None:
+            accepted_power = {}
+            for _, element in enumerate(self.all_element_names):
+                if self.active_s_parameters[element] is not None:
+                    operation = 1 - np.power(np.abs(self.active_s_parameters[element]), 2)
+                    accepted_power[element] = self.incident_power_element[element] * operation
+                else:
+                    accepted_power[element] = 0.0
+            total_accepted_power = sum(accepted_power.values())
+            return total_accepted_power
+        else:  # pragma: no cover
+            return None
+
+    @pyaedt_function_handler()
+    def __assign_weight_taper(self, a, b):  # pragma: no cover
+        """Assign weight to array.
+
+        Parameters
+        ----------
+        a : int
+            Index of array, column.
+        b : int
+            Index of array, row.
+
+        Returns
+        -------
+        float
+            Weight applied to specific index of the array.
+        """
+        taper = self.taper
+
+        if taper.lower() in ("flat", "uniform") or not self.__is_array:
+            return 1
+
+        cosinePow = 1
+        edgeTaper_dB = -200
+
+        edgeTaper = 10 ** ((float(edgeTaper_dB)) / 20)
+
+        threshold = 1e-10
+
+        # find the distance between current cell and array center in terms of index
+        lattice_vector = self.__lattice_vector[self.__freq_index]
+        if not lattice_vector or not len(lattice_vector) == 6:
+            return 1
```

Comment:
Not sure it makes sense to return 1 here, can you explain why ?

Reply:
@SMoraisAnsys This method is computing a number, that is called the weight of the element, which later is multiplied to the element field. If there isn't lattice vector (if it not an array), the weight can not be computer, and it is returning 1.0, because multiplying the field by 1 will not have any effect, I modified to 1.0 , because it is returning a float.